src/tools/database/milvus.ts

```diff
@@ -0,0 +1,290 @@
+        ),
+      collectionName: z
+        .string()
+        .optional()
+        .describe(
+          `The name of the collection to query, required for ${MilvusAction.GetCollectionInfo}, ${MilvusAction.Search}, ${MilvusAction.Insert}, and ${MilvusAction.Delete}`,
+        ),
+      vector: z
+        .array(z.number())
+        .optional()
+        .describe(`The vector to search for or insert, required for ${MilvusAction.Search}`),
+      vectors: z
+        .array(z.array(z.number()))
+        .optional()
+        .describe(`The vectors to insert, required for ${MilvusAction.Insert}`),
+      topK: z.coerce
+        .number()
+        .int()
+        .min(1)
+        .max(1000)
+        .default(10)
+        .optional()
+        .describe(
+          `The number of nearest neighbors to return for ${MilvusAction.Search}. Maximum is 1000`,
+        ),
+      filter: z
+        .string()
+        .optional()
+        .describe(`Optional filter expression for ${MilvusAction.Search}`),
+      metadata: z
+        .record(z.string(), z.any())
+        .optional()
+        .describe(`Additional metadata to insert with vectors for ${MilvusAction.Insert}`),
+      ids: z
+        .array(z.string().or(z.number()))
+        .optional()
+        .describe(`Array of IDs to delete for ${MilvusAction.Delete}`),
+      searchOutput: z
+        .array(z.string())
+        .optional()
+        .describe(`Fields to return in search results for ${MilvusAction.Search}`),
+    });
+  }
+
+  protected validateInput(
+    schema: AnyToolSchemaLike,
+    input: unknown,
+  ): asserts input is ToolInput<this> {
+    super.validateInput(schema, input);
+    if (input.action === MilvusAction.GetCollectionInfo && !input.collectionName) {
+      throw new ToolInputValidationError(
+        `Collection name is required for ${MilvusAction.GetCollectionInfo}, ${MilvusAction.Search}, ${MilvusAction.Insert}, and ${MilvusAction.Delete} actions.`,
+      );
+    }
+    if (input.action === MilvusAction.Search && (!input.collectionName || !input.vector)) {
+      throw new ToolInputValidationError(
+        `Both collection name and vector are required for ${MilvusAction.Search} action.`,
+      );
+    }
+    if (input.action === MilvusAction.Insert && (!input.collectionName || !input.vectors)) {
+      throw new ToolInputValidationError(
+        `Both collection name and vectors are required for ${MilvusAction.Insert} action.`,
+      );
+    }
+  }
+
+  static {
+    this.register();
+  }
+
+  public constructor(options: MilvusToolOptions) {
+    super(options);
+    if (!options.connection.address) {
+      throw new ValidationError([
+        {
+          message: `The address must be provide in the "host:port" format`,
+          propertyName: "connection.address",
+        },
+      ]);
+    }
```

Comment:
i noticed the linting failed once I removed the `public` keyword. For now I've added a rule to disable that line while eslint runs on the particularized file.

since the constructor does nothing once I remove this or only calls super(), is your recommendation to take it out entirely? WDYT? This behavior should be auto available anyways.